packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/data/ProjectSearchService.java

```diff
@@ -0,0 +1,205 @@
+		log.info("Project index: {}", elasticConfig.getProjectIndex());
+		return elasticConfig.getProjectIndex();
+	}
+
+	public String getAlias() {
+		log.info("Project alias: {}", elasticConfig.getProjectAlias());
+		return elasticConfig.getProjectAlias();
+	}
+
+	public String getCurrentAssetIndex() throws IOException {
+		return elasticService.getIndexFromAlias(getAlias());
+	}
+
+	@Data
+	public static class PermissionJoin {
+
+		private String name = null;
+		private UUID parent = null;
+
+		static PermissionJoin CreateProjectJoin() {
+			final PermissionJoin join = new PermissionJoin();
+			join.setName("project");
+			join.setParent(null);
+			return join;
+		}
+
+		static PermissionJoin CreatePermissionJoin(final UUID projectId) {
+			final PermissionJoin join = new PermissionJoin();
+			join.setName("permission");
+			join.setParent(projectId);
+			return join;
+		}
+	}
+
+	@Data
+	public static class ProjectDocument {
+
+		private final String type = "project";
+		private String userId; // the user id of who owns the project
+		private Boolean publicAsset; // whether the project is public or not
+		private Timestamp createdOn;
+		private Timestamp updatedOn;
+
+		// join field properties
+		private PermissionJoin permissionJoin;
+	}
+
+	@Data
+	public static class ProjectPermissionDocument {
+
+		private final String type = "permission";
+		private String userId; // the user id who holds the permission
+
+		// join field properties
+		private PermissionJoin permissionJoin;
+	}
+
+	/**
+	 * Setup the index and alias for the asset this service manages and ensure it is
+	 * empty
+	 *
+	 * @throws IOException If there is an error setting up the index and alias
+	 */
+	public void setupIndexAndAliasAndEnsureEmpty() throws IOException {
+		log.info("Setting up index {} and alias {}", getIndex(), getAlias());
+		final String index = getIndex();
+		try {
+			final String currentIndex = getCurrentAssetIndex();
+			// if another index exists, delete it.
+			if (!currentIndex.equals(index)) {
+				elasticService.deleteIndex(currentIndex);
+			}
+		} catch (final Exception e) {}
+		elasticService.createOrEnsureIndexIsEmpty(index);
+		elasticService.createAlias(index, getAlias());
+	}
+
+	/**
+	 * Teardown the index and alias for the asset this service manages
+	 *
+	 * @throws IOException If there is an error tearing down the index and alias
+	 */
+	public void teardownIndexAndAlias() throws IOException {
+		log.info("Tearing down index {}", getIndex());
+		final String index = getCurrentAssetIndex();
+		elasticService.deleteIndex(index);
+	}
+
+	public void indexProject(final Project project) throws IOException {
+		final ProjectDocument doc = new ProjectDocument();
+		doc.setUserId(project.getUserId());
+		doc.setPublicAsset(project.getPublicAsset());
+		doc.setCreatedOn(new Timestamp(System.currentTimeMillis()));
+		doc.setUpdatedOn(new Timestamp(System.currentTimeMillis()));
+		doc.setPermissionJoin(PermissionJoin.CreateProjectJoin());
```

Comment:
So this is the initial "permission" for a project, but it doesn't have a user assigned?  Why isn't the creator added to the list?

Reply:
That is the permission "join field" declaring that the document is a `"project"` document, and a parent in the parent-child relationship.